internal/pkg/skuba/kubernetes/node_version.go

```diff
@@ -127,14 +130,37 @@ func (nvi NodeVersionInfo) ToleratesClusterVersion(clusterVersion *version.Versi
 
 // AllNodesVersioningInfo returns the version info for all nodes in the cluster
 func AllNodesVersioningInfo(client clientset.Interface) (NodeVersionInfoMap, error) {
-	nodeList, err := client.CoreV1().Nodes().List(metav1.ListOptions{})
+	var lastErr error
+	var nodeList *v1.NodeList
+	err := wait.ExponentialBackoff(retry.DefaultBackoff, func() (bool, error) {
```

Comment:
Getting a list of Nodes is something that happens quite often on the code of skuba. Hence, maybe we can move this code into a proper function so this is helpful everywhere (as far as I could tell, the "retry on backoff" is not implemented everywhere).

Same applies for the listing of pods, I guess.

Reply:
I would prefer if we can do that at a later PR, since this PR is already very big. Would that be fine?

Reply:
Sure